### PR TITLE
Release 0.11.0

### PR DESCRIPTION
**Pull request by akaila-splunk**
- Update all dependencies to latest stable version and replace deprecated dependencies
- make library compatible with node version 10.0 and 14.0 and Splunk version 8.0 and 8.2.0
- Configured GitHub action for test and release
- Update library to use 'make up' and 'make test' convention

### CODE REVIEW

File: splunklogger.js [hunk 1]
```diff
@@ -14,7 +14,7 @@
  * under the License.
  */
 
-var request = require("request");
+var needle = require("needle");
```
Comment (fantavlik) [inline splunklogger.js:17]: - [ ] Let's add an item to the CHANGELOG.md for this - under `New features & APIs`: Replaced client implementation to use `needle` in lieu of deprecated `request`

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+name: Create Release
```
Comment (fantavlik) [inline .github/workflows/release.yml:1]: - [ ] Let's add an item to the CHANGELOG.md for this - under `Minor changes`: Converted CI from Travis to Github Actions and added CD

File: .travis.yml [hunk 3]
```diff
@@ -10,7 +10,7 @@ before_install:
     # Create .splunkrc file with default credentials
     - echo host=localhost >> $HOME/.splunkrc
     - echo username=admin >> $HOME/.splunkrc
-    - echo password=changeme >> $HOME/.splunkrc
+    - echo password=changed! >> $HOME/.splunkrc
     # Set SPLUNK_HOME
     - export SPLUNK_HOME="/opt/splunk"
     # Pull docker image
```
Comment (fantavlik) [inline .travis.yml:16]: Can we remove the .travis.yml file actually? Feel free to do that in this PR

**Review by fantavlik — CHANGES_REQUESTED** [review]
Looking good so far, I left comments in the review but two things to take care of:

- [ ] Need to add a new section to the CHANGELOG.md for this new release (notes on what to add to various sections are in the review below)
- [ ] Let's remove the .travis.yml since that's no longer used

File: splunklogger.js (hunk 1 above)
Comment (akaila-splunk) [reply splunklogger.js:17]: Updated CHANGELOG.md file as suggested.

File: .github/workflows/release.yml (hunk 2 above)
Comment (akaila-splunk) [reply .github/workflows/release.yml:1]: Updated CHANGELOG.md file as suggested.

File: .travis.yml (hunk 3 above)
Comment (akaila-splunk) [reply .travis.yml]: Removed .travis.yml flie.

**Comment by akaila-splunk** [comment]
> Looking good so far, I left comments in the review but two things to take care of:
> 
> * [ ]  Need to add a new section to the CHANGELOG.md for this new release (notes on what to add to various sections are in the review below)
> * [ ]  Let's remove the .travis.yml since that's no longer used

Updated CHANGELOG.md file as suggested and removed .travis.yml file.

**Review by fantavlik — APPROVED** [review]
Looking good, thanks again!

**Pull request merged**